docs/proposals/1374-mc-inference-gateways/README.md

```diff
@@ -0,0 +1,165 @@
+
+![Inference Architecture](images/gw-epp-ip.png)
+
+### Why Multi-Cluster?
+
+Until now, Inference Gateways have been focused exclusively on routing to a single cluster. Unfortunately, the resources needed to run LLM workloads continue to be scarce, and the desired capacity is rarely available within a single cluster. To address this, we propose expanding InferencePool to support multi-cluster routing.
+
+### Goals
+
+Enable Inference Gateways to route to backends in multiple clusters
+Follow a pattern that is familiar to users of Multi-Cluster Services (MCS) and/or Gateways
+
+### Non-Goals
+
+Be overly prescriptive about implementation details - this should focus on the resulting UX and leave significant flexibility in how it is achieved
+L4 ClusterIP routing and/or automatic DNS naming - all traffic needs to flow through the Inference Gateway for this pattern to be useful (otherwise the Endpoint Picker itself would be bypassed)
+
+## Proposal
+
+The multi-cluster Inference Gateway model will largely follow the multi-cluster services model, with a few key differences. We will omit DNS and ClusterIP resolution, and avoid a separate resource, e.g. ServiceExport, by inlining the concept within InferencePool. Additionally, we will add support for having separate Endpoint Pickers in each cluster.
+
+![InferencePoolImport](images/inference-pool-import.png)
+
+### API Changes
+
+#### InferencePool
+
+A new `inference.networking.k8s.io/export` annotation is added to InferencePool (replacement for ServiceExport resource in MCS). In the future this may become a field, but we’ll start with an annotation to allow for faster iteration. [We’ll avoid using a bool here to align with k8s API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types). The supported values to start will be `Local` and `ClusterSet`. In the future, we may allow for some intermediate values such as Regional or domain-prefixed values.
+
+#### InferencePoolImport
+
+A new API that mirrors ServiceImport from the MCS API. This allows anyone in a connected cluster to reference a Multi-Cluster InferencePool, even if the local cluster does not have a local InferencePool. In the context of Gateway API, that means that a Gateway could be configured to reference an InferencePoolImport, even if that cluster did not contain an InferencePool.
+This API will be used almost exclusively for tracking endpoints, but unlike MCS, we actually have two distinct sets of endpoints that we could track:
+
+1. Endpoint Pickers
+1. Model Server Endpoints
+
```

Comment:
We should also consider routing to remote EPPs through a remote cluster Gateway (see the original [design doc](https://docs.google.com/document/d/1QGvG9ToaJ72vlCBdJe--hmrmLtgOV_ptJi9D58QMD2w/edit?usp=sharing) appendix).

Reply:
>We should also consider routing to a remote EPP through a remote cluster Gateway (see the original [design doc](https://docs.google.com/document/d/1QGvG9ToaJ72vlCBdJe--hmrmLtgOV_ptJi9D58QMD2w/edit?usp=sharing) appendix).

Agree this should be possible - can this be a ~hidden or optional implementation detail?

>Why does InferencePoolImport need to know about the model server endpoints in a remote cluster?

It needs a way to support `FailOpen`, that does not strictly require awareness of model server endpoints in other clusters